### PR TITLE
Added rundll to rarun2

### PR DESCRIPTION
**Pull request by xarkes**
2017-07-31-114441_1055x522_scrot

Or on linux:

```
xarkes:~$ rarun2 rundll=/lib/libc-2.25.so rundll.fcn=system arg1="ls /"
bin   dev         etc   lib    lost+found  opt   root  sbin  sys  usr
boot  EFI_backup  home  lib64  mnt         proc  run   srv   tmp  var
```

As you can see on the screenshot, I am using a tiny hack to pass NULL to the function. I think I should add a format to the getStr() function in run.c so we can pass integers to the functions.

Also I am not very satisfied with the argument functions, I just did 4 to show as an example, should I add more or is there any generic way to handle this ?

### CODE REVIEW

File: libr/socket/run.c [hunk 1]
```diff
@@ -1016,5 +1037,43 @@ R_API int r_run_start(RRunProfile *p) {
 		exit (execv (p->_program, (char* const*)p->_args));
 #endif
 	}
+	if (p->_rundll) {
+		if (!p->_rundll_fcn) {
+			eprintf ("No function specified. Please set rundll.fcn\n");
+			return 1;
+		}
+		void *addr = r_lib_dl_open(p->_rundll);
```
Comment (radare) [inline libr/socket/run.c]: missing spaces before (

Comment (xarkes) [inline libr/socket/run.c]: damn it

File: libr/socket/run.c [hunk 2]
```diff
@@ -911,6 +916,22 @@ R_API int r_run_config_env(RRunProfile *p) {
 	return 0;
 }
 
+static void r_run_call1(void* fcn, void* arg1) {
```
Comment (XVilka) [inline libr/socket/run.c]: May be make those inline and move to some header? (Along with adding more parameters count).

**Comment by radare** [comment]
Please update manpage too

**Comment by radare** [comment]
squash and change commit message to contain "Fix #...."

**Comment by radare** [comment]
aka fix #8045 

https://github.com/radare/radare2/issues/8045

**Comment by radare** [comment]
also rename to runlib

**Comment by radare** [comment]
static inline

**Pull request merged**